packages/react/src/lazy.ts

```diff
@@ -0,0 +1,119 @@
+import { type ComponentType, type LazyExoticComponent, lazy as originalLazy } from 'react'
+import { noop } from './utils/noop'
+
+/**
+ * Creates a lazy function with custom default reload options
+ *
+ * @experimental This is experimental feature.
+ *
+ * @example
+ * ```tsx
+ * import { lazy } from '@suspensive/react'
+ *
+ * // Create a lazy factory with custom default reload count
+ * const customLazy = lazy.create({ reload: 5 })
+ *
+ * // Use the factory to create lazy components
+ * const Component1 = customLazy(() => import('./Component1'))
+ * const Component2 = customLazy(() => import('./Component2'))
+ *
+ * // Override default options for specific component
+ * const Component3 = customLazy(() => import('./Component3'), { reload: 2 })
+ * ```
+ */
+const createLazy = (defaultOptions: { reload: number }) => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const lazy = <T extends ComponentType<any>>(
+    load: () => Promise<{ default: T }>,
+    options?: { reload: number }
+  ): LazyExoticComponent<T> & {
+    load: () => Promise<void>
+  } => {
+    const finalOptions = options || defaultOptions
+
+    if (process.env.NODE_ENV === 'development' && finalOptions.reload) {
```

Comment:
I thought `finalOptions.reload` is always set. Is there a reason for this if condition?